### DISCUSSION TITLE
Apply discounts to a cart

### DISCUSSION THREAD

**Discussion by xV4L3x** [lunarphp/lunar#1354] [Q&A] [answered]
Hello everyone, I am trying to create a discount code section in my checkout page, when the user inserts the coupon, I query the database to search for discounts, my query looks like this:

`$discount = Discount::active()
            ->usable()
            ->where('coupon', $this->discountCode)->first();`
            
I have two questions at this point:
- Is there a function that looks into the cart lines and checks if the discount is valid for my cart products?
- How should I apply the discount to the cart? I have seen that there is a Facade called 'Discounts' that provides the apply() method, but I see that the field $discountBreakdown in the cart is needed, I don't understand what these breakdowns are and how they work

Thanks in advance for the replies

**Comment by SebastianBotez** [comment]
Have you managed to find out? I have the same question regarding how to apply discounts to the cart.

**Reply by xV4L3x** [reply, marked as answer]
Yes, I just add the coupon code to the cart and recalculate:

```php
$this->cart->coupon_code = $coupon;
$this->cart->save();
$this->cart->calculate();
```
